packages/infra/lib/ihe-gateway-v2-stack.ts

```diff
@@ -38,6 +44,54 @@ export class IHEGatewayV2LambdasNestedStack extends NestedStack {
     patientDiscoveryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentQueryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentRetrievalLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
+
+    this.addMetricFiltersAndAlarms(patientDiscoveryLambda, "PatientDiscoveryLambda", props, [
+      { filterPattern: "Aborted Error In Soap Response", threshold: 100 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 100 },
+    ]);
+
+    this.addMetricFiltersAndAlarms(documentQueryLambda, "DocumentQueryLambda", props, [
+      { filterPattern: "RegistryErrorList In Soap Response", threshold: 5 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 5 },
+    ]);
+
+    this.addMetricFiltersAndAlarms(documentRetrievalLambda, "DocumentRetrievalLambda", props, [
+      { filterPattern: "RegistryErrorList In Soap Response", threshold: 5 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 5 },
+    ]);
+  }
+
+  private addMetricFiltersAndAlarms(
+    lambdaFunction: Lambda,
+    functionName: string,
+    props: IHEGatewayV2LambdasNestedStackProps,
+    filters: { filterPattern: string; threshold: number }[]
+  ) {
+    const logGroup = lambdaFunction.logGroup;
+
+    filters.forEach(({ filterPattern, threshold }) => {
+      const sanitizedFilterPattern = filterPattern.replace(whitespaceRegex, "");
+      const metricFilter = logGroup.addMetricFilter(
+        `${functionName}-${sanitizedFilterPattern}-MetricFilter`,
+        {
+          metricNamespace: "IHEGatewayV2",
+          metricName: `${functionName}-${sanitizedFilterPattern}`,
+          filterPattern: FilterPattern.anyTerm(filterPattern),
+          metricValue: "1",
+        }
+      );
+
+      const alarm = metricFilter
+        .metric()
+        .createAlarm(this, `${functionName}-${sanitizedFilterPattern}-Alarm`, {
+          threshold,
+          evaluationPeriods: 1,
+          alarmDescription: `Alarm if ${functionName} encounters ${filterPattern}`,
+          treatMissingData: cloudwatch.TreatMissingData.NOT_BREACHING,
+        });
```

Comment:
I suggest we don't alert on these. CW-initiated alarms can't be "archived"/snoozed like we do with Sentry, so we would need to make a release to silent an alarm/update its configs.

We can instead add a widget to the prod dashboard and monitor once in a while as part of on-call (if not an auto report as indicated on the main review comment)?